compiler/src/iree/compiler/Codegen/Common/GPU/AMDGPUDistributeContract.cpp

```diff
@@ -292,10 +292,14 @@ struct DistributeContract final : OpDistributionPattern<vector::ContractionOp> {
         builder.create<vector::ShapeCastOp>(b.getLoc(), bVectorType, b);
     Value cCast =
         builder.create<vector::ShapeCastOp>(c.getLoc(), cVectorType, c);
-    FailureOr<Value> mmaOp = mmaKind.buildMmaOperation(
-        builder, loc, cVectorType, aCast, bCast, cCast);
-    assert(succeeded(mmaOp) && "Failed to construct mma op");
-    return builder.create<vector::ShapeCastOp>(c.getLoc(), c.getType(), *mmaOp);
+    SmallVector<Value> results;
+    LogicalResult createdMmaOp = mmaKind.buildUnderlyingOperations(
+        builder, loc, {aCast, bCast}, {cCast}, results);
+    if (failed(createdMmaOp)) {
+      llvm_unreachable("Should never fail to construct mma op");
+    }
```

Comment:
In prior discussions we generally avoid `llvm_unreachable` unless the code in question is critical. The potential downside of accidentally hitting this at some point in the future is not worth the benefit over an assert.
```suggestion
    [[maybe_unused]] LogicalResult createdMmaOp = mmaKind.buildUnderlyingOperations(
        builder, loc, {aCast, bCast}, {cCast}, results);
    assert(succeeded(createdMmaOp) && "Should never fail to construct mma op");
```